src/Bicep.Core/TypeSystem/ResourceType.cs

```diff
@@ -7,8 +7,8 @@ namespace Bicep.Core.TypeSystem
 {
     public class ResourceType : NamedObjectType
     {
-        public ResourceType(string name, IEnumerable<TypeProperty> properties, ITypeReference? additionalProperties, ResourceTypeReference typeReference)
-            : base(name, properties, additionalProperties)
+        public ResourceType(ResourceTypeReference typeReference, IEnumerable<TypeProperty> properties, ITypeReference? additionalProperties = null)
```

Comment:
>additionalProperties [](start = 119, length = 20)

nit - The name of this property bothers me given the preceding property also has `properties` in the name :( It really should be called `additionalPropertiesType` regardless of how redundant that becomes in the code.

Reply:
You asked for it, now you have to review a big old refactor :)

Reply:
Heh

---
In reply to: [493932954](https://github.com/Azure/bicep/pull/553#discussion_r493932954) [](ancestors = 493932954)